feature-libs/product-configurator/rulebased/core/state/reducers/configurator.reducer.ts

```diff
@@ -181,14 +187,37 @@ function takeOverChanges(
   action:
     | ConfiguratorActions.CreateConfigurationSuccess
     | ConfiguratorActions.ReadConfigurationSuccess
-    | ConfiguratorActions.UpdatePriceSummarySuccess
     | ConfiguratorActions.UpdateConfigurationFinalizeSuccess
     | ConfiguratorActions.ReadCartEntryConfigurationSuccess
     | ConfiguratorActions.ReadOrderEntryConfigurationSuccess,
   state: Configurator.Configuration
 ): Configurator.Configuration {
   const content = { ...action.payload };
   const groups = content.groups.length > 0 ? content.groups : state.groups;
+
+  const result = {
+    ...state,
+    ...content,
+    groups: groups,
+    interactionState: {
+      ...state.interactionState,
+      ...content.interactionState,
+      issueNavigationDone: true,
+    },
+  };
+  return result;
+}
+
+function takeOverPriceSupplementsChanges(
+  action: ConfiguratorActions.UpdatePriceSummarySuccess,
+  state: Configurator.Configuration
+): Configurator.Configuration {
+  const content = { ...action.payload };
+  const groups = ConfiguratorStateUtils.deepCopy(state.groups);
```

Comment:
That we need to discuss

Reply:
We should implement differently `ConfiguratorStateUtils.updateValuePrices`. Instead of being void function, it should return a new, fresh object. Moreover it should create a fresh object on whole levels of the change, according to the [immutable update pattern](https://redux.js.org/usage/structuring-reducers/immutable-update-patterns#correct-approach-copying-all-levels-of-nested-data). Thanks to that, only the updated branches of the state are copied, which is optimal. 

Deep copy of the whole object is not justified IMO, becasue:
  - we're also coping branches of the state that are not updated
  - it introduces a custom generic code for deep-copying objects, which requires additional good unit tests coverage 
  - btw. IMO it’s not the business of ConfiguratorStateUtils to deep copy objects (it's rather a generic util) 

Explanation, why we got a runtime error `ERROR TypeError: Cannot add property valuePrice, object is not extensible`:
To help developers, ngrx in [dev](https://github.com/ngrx/platform/blob/master/modules/store/src/runtime_checks.ts#L21) mode [freezes](https://github.com/ngrx/platform/blob/master/modules/store/src/meta-reducers/immutability_reducer.ts#L18) in runtime the objects of the state and actions -> to help identify accidental, unwanted mutations (which could lead in some cases to selectors' Observables not emitting a new value). For more, see https://ngrx.io/guide/store/configuration/runtime-checks